.github/workflows/test.yml

```diff
@@ -0,0 +1,71 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
```

Comment:
What happens if I can push some new code to my current branch and that branch is not master?

Reply:
the workflow is not going to run when changes are pushed to  any other branch different than master